packages/wallet-ui/src/components/ui/organism/AccountDetailsModal/AccountDetailsModal.view.tsx

```diff
@@ -1,38 +1,78 @@
+import { useState } from 'react';
 import {
   AccountImageDiv,
   AccountImageStyled,
+  AccountNameInput,
   AddressCopy,
   AddressQrCode,
   ButtonDiv,
   ButtonStyled,
+  EditIcon,
+  IconButton,
   Title,
   TitleDiv,
   Wrapper,
 } from './AccountDetailsModal.style';
 import { openExplorerTab } from 'utils/utils';
 import { useAppSelector } from 'hooks/redux';
 import { useMultiLanguage, useStarkNetSnap } from 'services';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPen, faCheck, faTimes } from '@fortawesome/free-solid-svg-icons';
 
 export const AccountDetailsModalView = () => {
   const networks = useAppSelector((state) => state.networks);
   const currentAccount = useAppSelector((state) => state.wallet.currentAccount);
-  const { getPrivateKeyFromAddress } = useStarkNetSnap();
+  const { getPrivateKeyFromAddress, setAccountName } = useStarkNetSnap();
   const { translate } = useMultiLanguage();
 
   const chainId = networks?.items[networks.activeNetwork]?.chainId;
   const address = currentAccount.address;
-  const addressIndex = currentAccount?.addressIndex ?? 0;
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [newAccountName, setNewAccountName] = useState(
+    currentAccount.accountName,
+  );
+
+  const handleSaveName = async () => {
+    if (
+      newAccountName.trim() &&
+      newAccountName !== currentAccount.accountName
+    ) {
+      await setAccountName(chainId, address, newAccountName);
+    }
+    setIsEditing(false);
+  };
+
   return (
     <div>
       <AccountImageDiv>
         <AccountImageStyled size={64} address={address} />
       </AccountImageDiv>
       <Wrapper>
         <TitleDiv>
-          <Title>
-            {translate('account')} {addressIndex + 1}
-          </Title>
-          {/* <ModifyIcon /> */}
+          {isEditing ? (
+            <>
+              <AccountNameInput
+                type="text"
+                value={newAccountName}
+                onChange={(e) => setNewAccountName(e.target.value)}
```

Comment:
```suggestion
                onChange={(event) => setNewAccountName(event.target.value)}
```

i guess you need to align, you use event at add account